compiler/rustc_resolve/src/late.rs

```diff
@@ -1963,13 +1962,13 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
             None,
             span,
         );
-        if let Some(LexicalScopeBinding::Res(res)) = binding { res != Res::Err } else { false }
+        matches!(binding, Some(LexicalScopeBinding::Res(res)) if res != Res::Err)
```

Comment:
Are you sure this is cleaner?

Reply:
It was cleaner before CI rightfully pointed out I'm an idiot. Now that the guard is there I'm not so sure anymore. Happy to drop those two hunks if you think it's better as is.

Reply:
I'm almost thinking that the following would be clearer...
```
match binding {
    Some(LexicalScopeBinding::Res(Res::Err)) => false,
    Some(LexicalScopeBinding::Res(_)) => true,
    _ => false,
}
```
I would just revert this chunk.

Reply:
Dropped.